kbase-extension/static/kbase/js/widgets/narrative_core/upload/stagingAreaViewer.js

```diff
@@ -600,37 +675,101 @@ define([
                 );
         },
 
+        renderImportButton: function() {
+
+            let importButton = $('<button></button>')
+            .addClass('kb-staging-table-import__button btn btn-xs btn-primary')
+            .text('Import Selected');
+
+            this.$elem.find('div.kb-staging-table-import').append(importButton);
+
+            /* 
+                By default import button is disabled until the user selects a data type 
+            */
+            this.disableImportButton();
+        },
+
+        disableImportButton: function() {
+
+            this.$elem.find('button.kb-staging-table-import__button')
+            .addClass('kb-staging-table-import__button__disabled')
+            .tooltip({
+                title: 'Select a file/s to continue.',
+                delay: {
+                    show: Config.get('tooltip').showDelay,
+                    hide: Config.get('tooltip').hideDelay
+                }
+            })
+            .off('click');
+        },
+
+        enableImportButton: function() {
+            let stagingAreaViewer = this; 
+
+            this.$elem.find('button.kb-staging-table-import__button')
+            .removeClass('kb-staging-table-import__button__disabled')
+            .tooltip('disable')
+            .unbind('click')
+            .on('click', function() {
+                stagingAreaViewer.initImport();
+            })
+            .on('keyPress', function() {
+                stagingAreaViewer.initImport();
+            });
+        },
+
+        initImport: function() {
+            let stagingAreaViewer = this; 
+
+            //get all of the selected checkbox file names and import type
+            $('input.kb-staging-table-body__checkbox-input:checked')
+            .each(function (i) {
```

Comment:
`i` is not used - can be removed

Reply:
hah good catch - cruft from when i was troubleshooting